SemVer/PartialVersion.cs

```diff
@@ -30,7 +30,7 @@ internal class PartialVersion
                     )?
                 )?
                 $",
-            RegexOptions.IgnorePatternWhitespace | RegexOptions.Compiled);
```

Comment:
Unfortunately, `RegexOptions.Compiled` is not supported in Mono/Xamarin. We could retain the optimisation for .NET4 using `#ifdefine` if you want; it would require adding another `.csproj` that has the define and have multiple libs in the `nuspec`.

The `RegexOptions.Compiled` isn't a significant performance improvement (http://blog.codinghorror.com/to-compile-or-not-to-compile/) for this library as it's unlikely you need to use `new SemVer.Version` excessively or in a tight loop.